packages/docmanager/src/widgetmanager.ts

```diff
@@ -268,15 +263,12 @@ class DocumentWidgetManager implements IDisposable {
         return true;
       }
       if (result) {
-        this._closeGuard = true;
-        widget.close();
-        this._closeGuard = false;
-        // Dispose of document widgets when they are closed.
         widget.dispose();
       }
       return result;
-    }).catch(() => {
+    }).catch(error => {
       widget.dispose();
+      throw error;
```

Comment:
should we rethrow or return `false`. Returning `undefined` is not an option because that makes the return type of the method `Promise<boolean | undefined>`